component-runtime-manager/src/test/java/org/talend/sdk/component/runtime/manager/service/record/RecordBuilderFactoryProviderTest.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (C) 2006-2022 Talend Inc. - www.talend.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.talend.sdk.component.runtime.manager.service.record;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.function.Function;
+
+import org.junit.jupiter.api.Test;
+import org.talend.sdk.component.api.service.record.RecordBuilderFactory;
+import org.talend.sdk.component.runtime.manager.ComponentManager;
+import org.talend.sdk.component.runtime.record.RecordBuilderFactoryImpl;
+
+public class RecordBuilderFactoryProviderTest {
```

Comment:
Should be better to enrich real test for [AvroRecordBuilderFactoryProvider](https://github.com/Talend/component-runtime/blob/master/component-runtime-beam/src/test/java/org/talend/sdk/component/runtime/beam/spi/AvroRecordBuilderFactoryProviderTest.java#L30) ? here, just testing a fake class.

To be complete, code 
```java
ofNullable(Thread.currentThread().getContextClassLoader())
                        .orElseGet(ClassLoader::getSystemClassLoader)
                        .loadClass("org.talend.sdk.component.runtime.beam.TalendIO");
```

can be put in a protected method :
```java
protected isTalendPipelineDesigner() {
    try {
        ofNullable(Thread.currentThread().getContextClassLoader())
                        .orElseGet(ClassLoader::getSystemClassLoader)
                        .loadClass("org.talend.sdk.component.runtime.beam.TalendIO");
        return true;
    } catch (final ClassNotFoundException | NoClassDefFoundError cnfe) {
        log
            .info(component-runtime-beam is not available, skipping AvroRecordBuilderFactory ({})",
              getClass().getName());
            return false;
    }
}
```
That make code clearer and easier to test

Reply:
outdated